packages/kbn-apm-config-loader/src/init_apm.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { loadConfiguration } from './config_loader';
+
+export const initApm = (
+  argv: string[],
+  rootDir: string,
+  isDistributable: boolean,
+  serviceName: string
+) => {
+  const apmConfigLoader = loadConfiguration(argv, rootDir, isDistributable);
+  const apmConfig = apmConfigLoader.getConfig(serviceName);
+
+  // we want to only load the module when effectively used
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const apm = require('elastic-apm-node');
```

Comment:
If I understood correctly, we use this same agent for the UI. However, this is the `elastic-apm-node` package, which is different from the `@elastic/apm-rum` used on the UI. 

Are we sure that we can reuse the same agent initialization for both: server and public?

Reply:
We're not using the same agent on server and browser, we're just sharing the same configuration and factorizing the configuration loading logic (the RUM config is then sent to the client in the injected metadatas). This code is only executed on the server-side.

Note that nothing changed in this PR in term of design/architecture, we just moved the script to a package to resolve the cyclic dependency between core and cli.